controllers/feeds/getFeedsV2.js

```diff
@@ -255,44 +320,26 @@ module.exports = async (req, res) => {
     }
     // get karma score for each post actor
     for (let i = 0; i < data.length; i++) {
-      if (data[i].latest_reactions?.downvotes) {
-        for (let j = 0; j < data[i].latest_reactions?.downvotes.length; j++) {
-          if (!postActors.includes(data[i].latest_reactions.downvotes[j].user_id)) {
-            postActors.push(data[i].latest_reactions.downvotes[j].user_id);
-          }
-        }
-      }
-      if (data[i].own_reactions?.downvotes) {
-        for (let j = 0; j < data[i].own_reactions?.downvotes.length; j++) {
-          if (!postActors.includes(data[i].own_reactions.downvotes[j].user_id)) {
-            postActors.push(data[i].own_reactions.downvotes[j].user_id);
-          }
-        }
-      }
+      const actor_from_reaction = getActorFromLatestReaction(data[i].latest_reactions);
+      postActors = [...postActors, ...actor_from_reaction];
     }
 
     const karmaScores = await UsersFunction.getUsersKarmaScore(User, postActors);
     for (let i = 0; i < data.length; i++) {
       const user = karmaScores.find((user) => user.user_id === data[i].actor.id);
       data[i].karma_score = roundingKarmaScore(user?.karma_score || 0);
-      if (data[i].latest_reactions?.downvotes) {
-        data[i].latest_reactions.downvotes = data[i].latest_reactions?.downvotes.map((downvote) => {
-          const user = karmaScores.find((user) => user.user_id === downvote.user_id);
-          return {
-            ...downvote,
-            karma_score: roundingKarmaScore(user?.karma_score || 0)
-          };
-        });
+
+      if (data[i].latest_reactions) {
+        data[i].latest_reactions = addKarmaScoreToLatestReaction(
+          data[i].latest_reactions,
+          karmaScores
+        );
       }
-      if (data[i].own_reactions?.downvotes) {
-        data[i].own_reactions.downvotes = data[i].own_reactions?.downvotes.map((downvote) => {
-          const user = karmaScores.find((user) => user.user_id === downvote.user_id);
-          return {
-            ...downvote,
-            karma_score: roundingKarmaScore(user?.karma_score || 0)
-          };
-        });
+
+      if (data[i].own_reactions) {
+        data[i].own_reactions = addKarmaScoreToLatestReaction(data[i].own_reactions, karmaScores);
       }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In this section, you are calling `getActorFromLatestReaction` and `addKarmaScoreToLatestReaction` functions in separate loops. This can be optimized by combining these operations into a single loop.

```diff
-  for (let i = 0; i < data.length; i++) {
-    const actor_from_reaction = getActorFromLatestReaction(data[i].latest_reactions);
-    postActors = [...postActors, ...actor_from_reaction];
-  }
-  const karmaScores = await UsersFunction.getUsersKarmaScore(User, postActors);
-  for (let i = 0; i < data.length; i++) {
+  const karmaScores = await UsersFunction.getUsersKarmaScore(User, []);
+  for (let i = 0; i < data.length; i++) {
+    const actor_from_reaction = getActorFromLatestReaction(data[i].latest_reactions);
+    postActors = [...postActors, ...actor_from_reaction];
+    karmaScores.push(...await UsersFunction.getUsersKarmaScore(User, actor_from_reaction));
     const user = karmaScores.find((user) => user.user_id === data[i].actor.id);
     data[i].karma_score = roundingKarmaScore(user?.karma_score || 0);
     if (data[i].latest_reactions) {
       data[i].latest_reactions = addKarmaScoreToLatestReaction(
         data[i].latest_reactions,
         karmaScores
       );
     }
     if (data[i].own_reactions) {
       data[i].own_reactions = addKarmaScoreToLatestReaction(data[i].own_reactions, karmaScores);
     }
   }
```
Please note that this change assumes `UsersFunction.getUsersKarmaScore` can handle an empty array as input and return an empty array as output. If this is not the case, you may need to add a condition to skip the call when `actor_from_reaction` is empty.


<!-- This is an auto-generated comment by OSS CodeRabbit -->